torch_geometric/utils/convert.py

```diff
@@ -119,12 +123,23 @@ def from_networkx(G):
     edge_index = torch.LongTensor(list(G.edges)).t().contiguous()
 
     data = {}
+    _, feat_dict = list(G.nodes(data=True))[0]
+    node_attributes = feat_dict.keys()
+    if group_node_attrs is all:
+        group_node_attrs = node_attributes
 
     for i, (_, feat_dict) in enumerate(G.nodes(data=True)):
+        if set(feat_dict.keys()) != set(node_attributes): raise ValueError(f'Not all nodes contain the same attributes')
         for key, value in feat_dict.items():
             data[str(key)] = [value] if i == 0 else data[str(key)] + [value]
 
+    _, _, feat_dict = list(G.edges(data=True))[0]
```

Comment:
I am not sure why this line is failing in the pytest tests, I can run it locally without fail...

Reply:
No worries about linting, I will fix that later :)

Reply:
This may be due to different `networkx` versions as they are changing their API frequently.

Reply:
Actually, it was just for a test of a graph with no edges, and I was looking in the wrong direction. So fortunately that's solved now!